src/modules/syncedVideoPlayer/SyncedVideoPlayer.ts

```diff
@@ -0,0 +1,78 @@
+import { Socket } from "socket.io-client";
+
+interface User {
+  id: string;
+  name: string;
+}
+
+class SyncedVideoPlayer {
+
+  /**
+   * User {id, name}
+   */
+  user: User
+
+  /**
+   * videoId - youtube videoId
+   */
+  videoId: String
+
+  /**
+   * channel - communication channel
+   */
+  channel: Socket
+
+  /**
+   * room - connected party socket room name
+   * Note: room and channel are the same thing in socket.io
+   */
+  room: String
+
+  /**
+   * player - Youtube player object
+   */
+  player: any
+
+  constructor(user: User, videoId: String, channel: Socket, room: String){
+    this.user = user;
+    this.videoId = videoId;
+    this.room = room;
+    this.channel = channel;
+
+    const id = 'some_id';
```

Comment:
Can we avoid hard-coding this?

Reply:
While we create the player it requires a videoId as per the docs.
I will have to check wether the object gets created without the video id.(if videoId is an optional param)

Reply:
This comment isn't about videoId, it's about line number 42 `const id = 'some_id';`

Reply:
Oh! I was using this for the videoId earlier, forgot that I had changed that.
 
This is for the DOM element id. 
`new window.YT.Player('player-${id}',.. ` this would still need a hardcoded id I think?

I would surely change `const id = 'some_id';` this, but what do we do about the `DOM element id`?

Reply:
I think we'll need to take it as a constructor param